web app/src/main/java/de/tinf15b4/kino/web/views/MovieView.java

```diff
@@ -58,75 +57,200 @@ public void enter(ViewChangeEvent event) {
                 if (!ratingsResponse.hasError()) {
                     List<RatedMovie> ratedMovies = Lists.newArrayList((RatedMovie[]) ratingsResponse.getValue());
 
                     if (ratedMovies.size() > 0) {
-                        GridLayout ratings = new GridLayout(4, 1);
-                        ratings.setMargin(true);
-                        ratings.setSpacing(true);
-                        ratings.setSizeFull();
-
-                        for (RatedMovie rm : ratedMovies) {
-                            SimpleDateFormat sdf = new SimpleDateFormat("dd.MM.yyyy HH:mm", Locale.GERMANY);
-                            ratings.addComponent(new Label(rm.getUser().getName()));
-                            ratings.addComponent(new Label(rm.getRating() + ""));
-                            ratings.addComponent(new Label(sdf.format(rm.getTime())));
-                            ratings.addComponent(new Label(rm.getDescription()));
-                        }
-
-                        Panel ratingsPanel = new Panel("Bewertungen", ratings);
-                        ratingsPanel.setId("ratingsPanel_" + m.getId());
-                        information.addComponent(ratingsPanel);
+                        VerticalLayout ratingsForm = createRatingsForm(ratedMovies);
+                        information.addComponent(ratingsForm);
                     }
+                }
+
+                // Playlist
+                RestResponse playlistResponse = userBean.getRestClient().getPlaylistForMovie(m.getId(), new Date(),
+                        new Date(new Date().getTime() + 1000L * 3600 * 24 * 7));
+                if (!playlistResponse.hasError()) {
+                    List<Playlist> playlistEntries = Lists.newArrayList((Playlist[]) playlistResponse.getValue());
+
+                    if (playlistEntries.size() > 0) {
+                        information.addComponent(createPlaylistForm(playlistEntries));
+                    }
+                }
+
+                information.setMargin(new MarginInfo(false, true));
+                content.addComponent(information);
+                content.setSizeFull();
+                content.setExpandRatio(information, 1f);
+                content.setMargin(true);
+                content.setSpacing(true);
+                this.addComponent(content);
+
+            }
+        }
+    }
+
+    private VerticalLayout createRatingsForm(List<RatedMovie> ratedMovies) {
+        VerticalLayout ratingsForm = new VerticalLayout();
+
+        Label heading = new Label("Bewertungen");
+        heading.setId("ratingsHeading");
+        ratingsForm.addComponent(heading);
+
+        for (RatedMovie rm : ratedMovies) {
+
+            ratingsForm.addComponent(createRatingEntry(rm));
+
+        }
+
+        ratingsForm.setId("ratingsForm");
+
+        return ratingsForm;
+    }
+
+    private VerticalLayout createRatingEntry(RatedMovie rm) {
```

Comment:
Suggestion: Wait a little while and see if the metrics tool catches it. Then we could use this as a refactoring for this week